demos/common/cpp/models/src/detection_model_retinaface.cpp

```diff
@@ -336,22 +339,23 @@ std::unique_ptr<ResultBase> ModelRetinaFace::postprocess(InferenceResult& infRes
     for (int idx = 0; idx < anchorCfg.size(); ++idx) {
         const auto bboxRaw = infResult.outputsData[separateOutputsNames[OT_BBOX][idx]];
         const auto scoresRaw = infResult.outputsData[separateOutputsNames[OT_SCORES][idx]];
-        const auto landmarksRaw = infResult.outputsData[separateOutputsNames[OT_LANDMARK][idx]];
         auto s = anchorCfg[idx].stride;
         auto anchorNum = anchorsFpn[s].size();
 
         auto validIndices = thresholding(scoresRaw, anchorNum, confidenceThreshold);
         filterScores(scores, validIndices, scoresRaw, anchorNum);
         filterBBoxes(bboxes, validIndices, bboxRaw, anchorNum, anchors[idx]);
-        filterLandmarks(landmarks, validIndices, landmarksRaw, anchorNum, anchors[idx], landmarkStd);
-
+        if (shouldDetectLandmarks) {
+            const auto landmarksRaw = infResult.outputsData[separateOutputsNames[OT_LANDMARK][idx]];
+            filterLandmarks(landmarks, validIndices, landmarksRaw, anchorNum, anchors[idx], landmarkStd);
+        }
         if (shouldDetectMasks) {
             const auto masksRaw = infResult.outputsData[separateOutputsNames[OT_MASKSCORES][idx]];
             filterMasksScores(masks, validIndices, masksRaw, anchorNum);
         }
     }
     // --------------------------- Apply Non-maximum Suppression ----------------------------------------------------------
-    auto keep = nms(bboxes, scores, boxIOUThreshold);
+    auto keep = nms(bboxes, scores, boxIOUThreshold, !shouldDetectLandmarks);
```

Comment:
Why does nms depend on landmarks presence? Is it just a way of distinguishing `ssh-mxnet` and `retinaface` and procedures they had for training? If so, please add a comment explaining that in code.

Reply:
In `nms` function for ssh-mxnet we should include boundaries in areas calculations, so, yes,  by passing `!shouldDetectLandmarks` we distinguish it from retinaface models .